embedded-graphics/src/fonts/mod.rs

```diff
@@ -107,200 +111,78 @@
-    };
-}
-
-/// Render text using the [`Font8x16`](./fonts/type.Font8x16.html) font
-///
-/// ```rust
-/// use embedded_graphics::{text_8x16, prelude::*, fonts::Font8x16, pixelcolor::Rgb565};
-///
-/// let text: Font8x16<Rgb565> = text_8x16!("Hello world!");
-/// let styled_text: Font8x16<Rgb565> = text_8x16!(
-///     "Hello world!",
-///     text_color = Some(Rgb565::RED),
-///     background_color = Some(Rgb565::GREEN)
-/// );
-/// ```
-///
-/// Style properties like `text_color` map to properties in the [`TextStyle`] struct.
-///
-/// [`TextStyle`]: style/struct.TextStyle.html
-#[macro_export]
-macro_rules! text_8x16 {
-    ($text:expr $(, $style_key:ident = $style_value:expr )* $(,)?) => {
-        $crate::impl_text!(Font8x16, $text $(, $style_key = $style_value )*)
-    };
-}
-
-/// Render text using the [`Font12x16`](./fonts/type.Font12x16.html) font
-///
-/// ```rust
-/// use embedded_graphics::{text_12x16, prelude::*, fonts::Font12x16, pixelcolor::Rgb565};
-///
-/// let text: Font12x16<Rgb565> = text_12x16!("Hello world!");
-/// let styled_text: Font12x16<Rgb565> = text_12x16!(
-///     "Hello world!",
-///     text_color = Some(Rgb565::RED),
-///     background_color = Some(Rgb565::GREEN)
-/// );
-/// ```
-///
-/// Style properties like `text_color` map to properties in the [`TextStyle`] struct.
-///
-/// [`TextStyle`]: style/struct.TextStyle.html
-#[macro_export]
-macro_rules! text_12x16 {
-    ($text:expr $(, $style_key:ident = $style_value:expr )* $(,)?) => {
-        $crate::impl_text!(Font12x16, $text $(, $style_key = $style_value )*)
-    };
-}
-
-/// Render text using the [`Font24x32`](./fonts/type.Font24x32.html) font
-///
-/// ```rust
-/// use embedded_graphics::{text_24x32, prelude::*, fonts::Font24x32, pixelcolor::Rgb565};
-///
-/// let text: Font24x32<Rgb565> = text_24x32!("Hello world!");
-/// let styled_text: Font24x32<Rgb565> = text_24x32!(
-///     "Hello world!",
-///     text_color = Some(Rgb565::RED),
-///     background_color = Some(Rgb565::GREEN)
-/// );
-/// ```
-///
-/// Style properties like `text_color` map to properties in the [`TextStyle`] struct.
-///
-/// [`TextStyle`]: style/struct.TextStyle.html
-#[macro_export]
-macro_rules! text_24x32 {
-    ($text:expr $(, $style_key:ident = $style_value:expr )* $(,)?) => {
-        $crate::impl_text!(Font24x32, $text $(, $style_key = $style_value )*)
-    };
-}
-
 #[cfg(test)]
 mod tests {
     use super::*;
     use crate::pixelcolor::{BinaryColor, Rgb565, RgbColor};
+    use crate::style::{Styled, TextStyle};
 
     #[test]
     fn font_macros() {
-        let _text: Font6x8<BinaryColor> = text_6x8!("Hello!");
-        let _text: Font6x12<BinaryColor> = text_6x12!("Hello!");
-        let _text: Font8x16<BinaryColor> = text_8x16!("Hello!");
-        let _text: Font12x16<BinaryColor> = text_12x16!("Hello!");
-        let _text: Font24x32<BinaryColor> = text_24x32!("Hello!");
+        let _text: Styled<Text<'_>, TextStyle<BinaryColor, Font6x8>> =
```

Comment:
Ouch, I hope end users don't have to type this out very often :sweat_smile:

Reply:
Yes, the `Styled<...>` types for texts and also primitives are pretty long. If this turns out to be a problem we could add type aliases to shorten these types later:
```rust
type StyledText<'a, C, F> = Styled<Text<'a>, TextStyle<C, F>>;
let _text: StyledText<'_, BinaryColor, Font6x8> = egtext!("Hello!", font = Font6x8);
```

Reply:
Ah yeah that's true. Might not ever be a problem so let's see what community feedback we get over time.